### PR TITLE
fix(cmake): add `Find(GTest)` for mock libraries

### LINKED ISSUE #14432: Mock libraries need to find GTest
@coryan:

> I think it is a bug in `google-cloud-cpp`.  This code:
> 
> https://github.com/googleapis/google-cloud-cpp/blob/b5026cc636b32fa045c1b6868e94151dffbdd061/cmake/GoogleCloudCppLibrary.cmake#L350-L352
> 
> Assumes BUILD_TESTING is defined and GMock is found via:
> 
> https://github.com/googleapis/google-cloud-cpp/blob/b5026cc636b32fa045c1b6868e94151dffbdd061/CMakeLists.txt#L241-L244

I agree. I think the fix is to instead say:

```
if (BUILD_TESTING OR GOOGLE_CLOUD_CPP_WITH_MOCKS)
    include(FindGMockWithTargets)
endif ()
```

> we need to figure out [...] why it was missed in our testing, and then probably add a patch to get this release pushed through conda.

`cmake-install` builds with `BUILD_TESTING=OFF` (and implied `GOOGLE_CLOUD_CPP_WITH_MOCKS=ON`). It also checks that the mocks are installed by linking an application against them.

The application doesn't `#include <gmock/gmock.h>`, or use any GTest types, which I think has hidden the problem.

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #14432 

`NOT BUILD_TESTING AND GOOGLE_CLOUD_CPP_WITH_MOCKS` is a valid configuration of the library. In this case, we need to make sure to `Find(GTest)` before trying to link the mocking libraries against GTest.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14433)

### CODE REVIEW

**Comment by coryan** [comment]
Close, but no cigar, you also need:

```patch
diff --git a/cmake/FindGMockWithTargets.cmake b/cmake/FindGMockWithTargets.cmake
index 63bb8c86cf..691042883f 100644
--- a/cmake/FindGMockWithTargets.cmake
+++ b/cmake/FindGMockWithTargets.cmake
@@ -93,15 +93,16 @@ function (google_cloud_cpp_transfer_library_properties target source)
 endfunction ()
 
 include(CTest)
+find_package(GTest CONFIG QUIET)
+find_package(GMock CONFIG QUIET)
+
 if (TARGET GTest::gmock)
     # GTest::gmock is already defined, do not define it again.
-elseif (NOT BUILD_TESTING)
+elseif (NOT BUILD_TESTING AND NOT GOOGLE_CLOUD_CPP_WITH_MOCKS)
     # Tests are turned off via -DBUILD_TESTING, do not load the googletest or
     # googlemock dependency.
 else ()
     # Try to find the config package first. If that is not found
-    find_package(GTest CONFIG QUIET)
-    find_package(GMock CONFIG QUIET)
     if (NOT GTest_FOUND)
         find_package(GTest MODULE REQUIRED)
```

**Comment by coryan** [comment]
Consider using this for local testing:

```patch
diff --git a/ci/cloudbuild/builds/cmake-split-install.sh b/ci/cloudbuild/builds/cmake-split-install.sh
index b9f8e264ea..9be49a1fa9 100755
--- a/ci/cloudbuild/builds/cmake-split-install.sh
+++ b/ci/cloudbuild/builds/cmake-split-install.sh
@@ -38,6 +38,11 @@ install_args=(
   -DBUILD_SHARED_LIBS=ON
 )
 
+curl -fsSL https://github.com/Kitware/CMake/releases/download/v3.30.0/cmake-3.30.0-linux-x86_64.tar.gz | 
+  tar -C $HOME -xzf -
+
+export PATH=$HOME/cmake-3.30.0-linux-x86_64/bin:$PATH
+
 io::log_h2 "Building and installing common libraries"
 # We need a custom build directory
 mapfile -t core_cmake_args < <(cmake::common_args cmake-out/common-libraries)

```

**Comment by dbolduc** [comment]
> Close, but no cigar, you also need:

Good catch. Thanks.

> Consider using this for local testing:

Thanks, this is what I needed. I reverted the changes in `ci/verify_current_targets/*` because they weren't doing anything.

We are still not exercising CMake >= 30 in the CI.

**Review by coryan — APPROVED** [review]

**Review by coryan — APPROVED** [review]

File: ci/verify_current_targets/CMakeLists.txt [hunk 1]
```diff
@@ -50,7 +50,7 @@ find_package(Threads REQUIRED)
 include(CTest)
 
 function (add_test_case name)
-    add_executable("${name}" verify_current_targets.cc)
+    add_executable("${name}" "verify_current_targets.cc")
```
[the remark this answers is not in the document]
Comment (dbolduc) [reply ci/verify_current_targets/CMakeLists.txt]: Done

**Pull request merged**